interface/repository/rest_pokemon_repository.go

```diff
@@ -0,0 +1,65 @@
+package repository
+
+import (
+	"encoding/json"
+	"errors"
+	"io/ioutil"
+	"net/http"
+	"strconv"
+
+	"github.com/FernandoGal25/academy-go-q42021/domain/model"
+)
+
+/*
+	Repository that handles the operations of pokemon in a external REST API.
+*/
+type RESTPokemonRepository struct {
+	APIGateway string
+}
+
+func NewRestPokemonRepository(p string) RESTPokemonRepository {
+	return RESTPokemonRepository{p}
+}
+
+func (r RESTPokemonRepository) FindByID(ID uint64) (*model.Pokemon, error) {
```

Comment:
Just for curiosity, why use uint64, if you will be converting that value to int?